crates/hyperdrive-math/src/base.rs

```diff
@@ -57,15 +55,15 @@ impl State {
         &self,
         scaled_maturity_time: FixedPoint,
         current_time: U256,
-    ) -> FixedPoint {
+    ) -> Result<FixedPoint> {
         let scaled_latest_checkpoint =
-            FixedPoint::from(self.to_checkpoint(current_time)) * fixed!(1e36);
-        let scaled_position_duration = self.position_duration() * fixed!(1e36);
+            FixedPoint::from(self.to_checkpoint(current_time)).mul_down(fixed!(1e36))?;
```

Comment:
did we lose the `*` sugar or are you preferring `mul_down`? We've been somewhat inconsistent with this, but imo the sugar operators (e.g. `*`) are better unless we're explicitly trying to match a solidity function because they are more readable.

Reply:
Reverted for now 👍